openvdb/openvdb/tree/NodeUnion.h

```diff
@@ -44,42 +42,28 @@ class NodeUnion
     const ValueT& getValue() const { return mValue; }
     ValueT& getValue() { return mValue; }
     void setValue(const ValueT& val) { mValue = val; }
-};
 
-
-// Template specialization for values of POD types (int, float, pointer, etc.)
-template<typename ValueT, typename ChildT>
-class NodeUnion<ValueT, ChildT, typename std::enable_if<std::is_pod<ValueT>::value>::type>
-{
-private:
-    union { ChildT* mChild; ValueT mValue; };
-
-public:
-    NodeUnion(): mChild(nullptr) {}
-
-    ChildT* getChild() const { return mChild; }
-    void setChild(ChildT* child) { mChild = child; }
-
-    const ValueT& getValue() const { return mValue; }
-    ValueT& getValue() { return mValue; }
-    void setValue(const ValueT& val) { mValue = val; }
+    // Small check to ensure this class isn't
+    // selected for some expected types
+    static_assert(!ValueTraits<ValueT>::IsVec &&
+        !ValueTraits<ValueT>::IsMat &&
+        !ValueTraits<ValueT>::IsQuat &&
+        !std::is_same<ValueT, math::Coord>::value &&
+        !std::is_arithmetic<ValueT>::value,
+        "Unexpected instantiation of NodeUnion");
```

Comment:
I'll probably remove this `static_assert` as it isn't really doing the right job - I wanted a check to ensure the right specialization was being selected, not the wrong one